includes/classes/class.base.php

```diff
@@ -85,21 +85,35 @@ function notify($eventID, $param1 = array(), & $param2 = NULL, & $param3 = NULL,
       error_log( strftime("%Y-%m-%d %H:%M:%S") . ' [main_page=' . $main_page . '] ' . $eventID . $output . "\n", 3, $file);
     }
 
-    // handle observers
-    // observers can fire either a generic update() method, or a notifier-point-specific updateNotifierPointCamelCased() method. The specific one will fire if found; else the generic update() will fire instead.
-    $observers = & base::getStaticObserver();
-    if (is_null($observers)) {
-      return;
-    }
+      // handle observers ("listeners")
+      // observers will fire in in this order:
+      // a snake_cased method name matching the notifier like: notifier_point_in_lower_case()
+      // a camelCased method named after the notifer like: updateNotifierPointCamelCased()
+      // else the generic update() (and it should parse $eventID if it is intended to handle multiple attached notifiers)
+      $observers = & base::getStaticObserver();
+      if (is_null($observers)) {
+          return;
+      }
 
-      foreach($observers as $key=>$obs) {
-        if ($obs['eventID'] == $eventID || $obs['eventID'] === '*') {
-         $method = 'update';
-         $testMethod = $method . self::camelize(strtolower($eventID), TRUE);
-         if (method_exists($obs['obs'], $testMethod))
-           $method = $testMethod;
-         $obs['obs']->{$method}($this, $eventID, $param1,$param2,$param3,$param4,$param5,$param6,$param7,$param8,$param9);
-        }
+      foreach ($observers as $key => $obs) {
+          if ($obs['eventID'] == $eventID || $obs['eventID'] === '*') {
+
+              // First we check for a snake_cased method name of the notifier Event, IF IT STARTS WITH "notify_" or "notifier_"
+              // If found, we call it, but the $eventID is NOT passed as 2nd param, unlike the updateX() calls below
+              $snake_case_method = strtolower($eventID);
+              if (preg_match('/^notif(y|ier)_/', $snake_case_method) && method_exists($obs['obs'], $snake_case_method)) {
+                  $obs['obs']->{$snake_case_method}($this, $param1, $param2, $param3, $param4, $param5, $param6, $param7, $param8, $param9);
+                  return;
+              }
+              // If the first check failed, we check for a camelCased version starting with "update": updateNotifierNameCamelCased()
+              // If found, we call it. For legacy reasons we include $eventID as the 2nd parameter.
+              // If the camelCased version is not found, we call "update()". If it doesn't exist then a PHP fatal error will occur.
+              $method = 'update';
+              if (method_exists($obs['obs'], $camelCaseMethod = $method . self::camelize(strtolower($eventID), true))) {
+                  $method = $camelCaseMethod;

```

Comment:
haven't tested this, but I thought that you could only assign a variable to an argument if it is passed by reference.

Reply:
This is just really hard to read.  Could we not just do 

```
$camelCaseMethod = $method . self::camelize(strtolower($eventID), true); 
if (method_exists($obs['obs'], $camelCaseMethod)) {
   $method = $camelCaseMethod;
}


```

Reply:
This part is pre-existing code. The pass-by-reference works when *received* by reference in the observer class.